diff --git a/tests.cpp b/tests.cpp
index e3a283c..e9adbe1 100644
--- a/tests.cpp
+++ b/tests.cpp
@@ -9,6 +9,16 @@
 
 #include "tests.h"
 
+int etape=0;
+
+// Miscellanous functions {{{
+void step(string msg, int *i=&etape,bool init=false) {
+  *i = init ? 0 : *i+1;
+  cout << "Etape " << *i << ": " << msg << endl;
+}
+
+// }}}
+
 // Tests sur les formes geometriques {{{
 void Tests::tests_unitaires_formes() {
   cout << "----- Tests des formes geometriques -----\n";
@@ -197,6 +207,74 @@ void Tests::tests_unitaires_couche() {
 
 // Tests sur la classe Canevas {{{
 void Tests::tests_unitaires_canevas() {
+  cout << "----- Tests de la classe Canevas -----\n\n";
+
+  cout << "--- Initialisation d'un canevas ---\n";
+  Canevas c1;
+  cout << "Couches: ";
+  c1.getEtats(cout);
+  cout << "Affichage: {\n";
+  c1.afficher(cout);
+  cout << "}\n\n";
+
+  cout << "--- Ajout d'une forme (doit aller dans la couche 0) ---\n";
+  cout << "Couches: ";
+  c1.getEtats(cout);
+  cout << "Reussite: " << (c1.ajouterForme(new Carre())? "Oui":"Non")
+       << "\nAffichage: {\n";
+  c1.afficher(cout);
+  cout << "}\n\n";
+
+  cout << "--- Activation de la couche 3 ---\n";
+  cout << "Reussite: " << (c1.activerCouche(3)? "Oui":"Non")
+       << "\nCouches: ";
+  c1.getEtats(cout);
+  cout << "Affichage: {\n";
+  c1.afficher(cout);
+  cout << "}\n\n";
+
+  cout << "--- Ajout de forme (doit aller dans la couche 3) ---\n";
+  cout << "Reussite: " << (c1.ajouterForme(new Carre())? "Oui":"Non")
+       << "\nCouches: ";
+  c1.getEtats(cout);
+  cout << "Affichage: {\n";
+  c1.afficher(cout);
+  cout << "}\n\n";
+
+  cout << "--- Reinitialisation de la couche active (3) ---\n";
+  cout << "Reussite: " << (c1.reinitialiserCouche(3)? "Oui":"Non")
+       << "\nCouches: ";
+  c1.getEtats(cout);
+  cout << "Affichage: {\n";
+  c1.afficher(cout);
+  cout << "}\n\n";
+
+  cout << "--- Translation de couche initialisée (couche 3 par 1,2) ---\n";
+  cout << "Reussite: " << (c1.translater(1, 2)? "Oui":"Non")
+       << "\nCouches: ";
+  c1.getEtats(cout);
+  cout << "Affichage: {\n";
+  c1.afficher(cout);
+  cout << "}\n\n";
+
+  cout << "--- Translation d'une couche active (couche 0 par 1,2) ---\n";
+  c1.activerCouche(0);
+  cout << "Reussite: " << (c1.translater(1, 2)? "Oui":"Non")
+       << "\nCouches: ";
+  c1.getEtats(cout);
+  cout << "Affichage: {\n";
+  c1.afficher(cout);
+  cout << "}\n\n";
+
+  cout << "--- Réinitialisation du canevas ---\n";
+  cout << "Reussite: " << (c1.reinitialiser()? "Oui":"Non")
+       << "\nCouches: ";
+  c1.getEtats(cout);
+  cout << "Affichage: {\n";
+  c1.afficher(cout);
+  cout << "}\n";
+
+
 }; // }}}
 
 // Execution de tout les tests unitaires {{{
@@ -217,106 +295,7 @@ void Tests::tests_application() {
 
 // Premiere application {{{
 void Tests::tests_application_cas_01() {
-  // Mise en place
-  int etape = 1;
   Canevas c;
-
-  cout << "TESTS APPLICATION (CAS 01)" << endl;
-  // Il faut ajouter les operations realisant ce scenario de test.
-  //
-  cout << "Etape " << etape++
-       << ": Activation de la couche 1" << endl;
-  c.activerCouche(1);
-
-  cout << "Etape " << etape++
-       << ": Ajout des trois formes géométriques suivantes" << endl;
-  cout << "\t* Un rectangle (x=0, y=0, largeur=2, hauteur=3)" << endl;
-  c.ajouterForme(new Rectangle(0,0,2,3));
-  cout << "\t* Un carré (x=2, y=3, cote=4)" << endl;
-  c.ajouterForme(new Carre(2,3,4));
-  cout << "\t* Un cercle (x=7, y=8, rayon=6)" << endl;
-  c.ajouterForme(new Cercle(7,8,6));
-
-  cout << "Etape " << etape++
-       << ": Activer la couche 2" << endl;
-  c.activerCouche(2);
-
-  cout << "Etape " << etape++
-       << ": Ajouter la forme géométrique suivante" << endl;
-  cout << "\t* Un rectangle (x=0, y=0, largeur=4, hauteur=5)" << endl;
-  c.ajouterForme(new Rectangle(0,0,4,5));
-
-  cout << "Etape " << etape++
-       << ": Afficher le canevas" << endl;
-  c.afficher(cout);
-
-  cout << "Etape " << etape++
-       << ": Afficher l'aire du canevas" << endl;
-  cout << "\t* Aire du canevas: " << c.aire() << endl;
-
-  cout << "Etape " << etape++
-       << ": Activer la couche 0 et ajouter les formes suivantes" << endl;
-  c.activerCouche(0);
-  cout << "\t* Un rectangle (x=0, y=0, largeur=1, hauteur=1)" << endl;
-  c.ajouterForme(new Rectangle());
-  cout << "\t* Un carré (x=0, y=0, cote=1)" << endl;
-  c.ajouterForme(new Carre());
-  cout << "\t* Un cercle (x=0, y=0, rayon=1)" << endl;
-  c.ajouterForme(new Cercle());
-
-  cout << "Etape " << etape++
-       << ": Translater les formes de la couche selon x=5, y=5 pour obtenir les formes suivantes lorsque affiché" << endl;
-  c.translater(5, 5);
-  cout << "\t* Un rectangle (x=5, y=5, largeur=1, hauteur=1)" << endl;
-  cout << "\t* Un carré (x=5, y=5, cote=1)" << endl;
-  cout << "\t* Un cercle (x=5, y=5, rayon=1)" << endl;
-  c.afficher(cout);
-
-
-
-  cout << "Etape " << etape++
-       << ": Couche 2 - initialisée" << endl;
-  c.reinitialiserCouche(2);
-
-  cout << "Etape " << etape++
-       << ": Couche 3 - initialisée" << endl;
-  c.reinitialiserCouche(3);
-
-  cout << "Etape " << etape++
-       << ": Couche 4 - initialisée" << endl;
-  c.reinitialiserCouche(4);
-
-  cout << "Etape " << etape++
-       << ": Afficher le canevas" << endl;
-  c.afficher(cout);
-
-  cout << "Etape " << etape++
-       << ": Afficher l'aire du canevas" << endl;
-
-  cout << "Etape " << etape++
-       << ": Retirer la première forme de la couche 1" << endl;
-  c.activerCouche(1);
-  c.retirerForme(0);
-
-  cout << "Etape " << etape++
-       << ": Afficher le canevas" << endl;
-  c.afficher(cout);
-
-  cout << "Etape " << etape++
-       << ": Afficher l'aire du canevas" << endl;
-  cout << "\t* Aire du canevas: " << c.aire() << endl;
-
-  cout << "Etape " << etape++
-       << ": Réinitialiser le canevas" << endl;
-  c.reinitialiser();
-
-  cout << "Etape " << etape++
-       << ": Afficher le canevas" << endl;
-  c.afficher(cout);
-
-  cout << "Etape " << etape++
-       << ": Afficher l'aire du canevas" << endl;
-  cout << "\t* Aire du canevas: " << c.aire() << endl;
 }; // }}}
 
 // Deuxieme application {{{
